src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java

```diff
@@ -396,7 +399,7 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
     private <T> boolean isCrossPartitionQuery(@NonNull DocumentQuery query, @NonNull Class<T> domainClass) {
         final Optional<String> partitionKeyName = getPartitionKeyField(domainClass);
 
-        if (!partitionKeyName.isPresent()) {
+        if (!partitionKeyName.isPresent() || query == null) {
```

Comment:
looks we can remove `query == null` if there is `ALL` type.

Reply:
Will update in another PR.